### PR TITLE
ls: Give non-zero exit code when an unknown resource is specified

### PR DESCRIPTION
**Pull request by kyleam**
This allows a command-line caller to use `reproman ls` to check if a resource exists, as described at gh-521.

### CODE REVIEW

File: reproman/interface/ls.py [hunk 1]
```diff
@@ -100,4 +109,10 @@ def __call__(resrefs=None, verbose=False, refresh=False):
             manager.save_inventory()
         else:
             ui.message('Use --refresh option to view updated status.')
+
+        if unknown_resrefs:
+            raise ResourceNotFoundError(
+                "Could not find the following resources: {}"
+                .format(unknown_resrefs))
```
Comment (yarikoptic) [inline reproman/interface/ls.py:116]:
```suggestion
                .format(' '.join(unknown_resrefs)))
```

Comment (kyleam) [reply reproman/interface/ls.py:116]: I'd prefer to just format the list as is, but given you feel strongly enough to suggest a `.join`, I'll make that change.  I think space is a bad delimiter though.  I'll go with ', '.

**Comment by kyleam** [comment]
> Also pr description just says "as described at [gh-521](https://github.com/ReproNim/reproman/issues/521)" - isn't it also closing it?

The second commit has a "Closes ..." statement.

**Comment by yarikoptic** [comment]
Great, thank you. Let's merge when turns green

**Pull request merged**